### ISSUE TITLE
"Real" Time Traveling

### ISSUE THREAD

**Issue by eccentricdevotion** [eccentricdevotion/TARDIS#305]
_Transferred from Bukkit Dev 1182
user_10479419 created this issue Oct 20, 2015_

One of the most important characteristics of the TARDIS in the show, which we don't have in this plugin, is that it can travel in time. Well, time travel in a game like Minecraft is as impossible as in the real life, so I do not blame you for not have implemented time travel yet :P

But maybe time travel to the past is not as impossible... if we just record every change in every world and every movement of every player! well, I know that is not gonna happen. It's useless, it would consume resources vastly, and would be difficult to code. But, there is an alternative, not as cool, but still cool.

For example, the staff of a server want to make the planet Trenzalore, but the two versions of it, the past and the future. Actually they could just make two worlds, called Trenzalore_Past and Trenzalore_Future, and let the player push the random location button until he gets the desired version of the planet.

Or, time traveling could be implemented, and the player just gets Trenzalore as location, and then with another control, he can time travel to the version of it he wants!

So, in the config of the plugin, worlds would be arranged something like this:

- Spawn
-- Spawn_2013
-- Spawn_2014
- Trenzalore
-- Trenzalore_Future

Worlds with just one dash would be the present, the default world to go. And just if the player wants to time travel to a different version of it, he uses the time travel control, and selects one of the worlds with two dashes.

Basically, worlds that the staff previously made, as subworlds of the main worlds. I think that this is easy to do, and its possibilities would be unlimited, fantastic and totally cool.

**Comment by eccentricdevotion** [comment]
Don't really need a double dash, just naming the world with the `_Future` suffix would be sufficient
Or

```
- -Trenzalore #past world
- Trenzalore #present world
- +Trenzalore #future world
```

**Comment by eccentricdevotion** [comment]
Would it be worth allowing server admins to attach TARDIS areas to this system as well? I'm thinking past, present, and future versions of a specific city, each version built within the same world to conserve server resources. An admin might want to set up one world to store all these different time creations.
Sure, area disks accomplish traveling to these areas separately, but when each area is meant to be the same physical location across different times, the disks seem to spoil the experience. Hiding the area disks in this situation and replacing them with this repeater control makes it a more believable experience.

**Comment by FamousL** [comment]
CoreProtect data may help out here, being able to tell when a block was broken/pipped into being and where could make for some interesting cases though.
